gwcs/wcs.py

```diff
@@ -519,31 +519,65 @@ def __repr__(self):
             self.output_frame, self.input_frame, self.forward_transform)
         return fmt
 
-    def footprint(self, bounding_box=None, center=False):
+    def footprint(self, bounding_box=None, center=False, kind="all"):
```

Comment:
Not sure if `kind` is very descriptive parameter.  Could it be something like `type` or `axis_type`?  When I first saw `kind`, thought of it as an adjective.  I.e. "compute the footprint nicely."  😆